applications/StructuralMechanicsApplication/custom_constitutive/linear_j2_plasticity_3d.cpp

```diff
@@ -0,0 +1,464 @@
+              (2 * mu * theta_new_b * (N_new(0) * N_new(2)));
+    D(0, 3) = -(2 * mu * theta_new_b * (N_new(0) * N_new(3)));
+    D(0, 4) = -(2 * mu * theta_new_b * (N_new(0) * N_new(4)));
+    D(0, 5) = -(2 * mu * theta_new_b * (N_new(0) * N_new(5)));
+
+    D(1, 0) = volumetric_modulus + (2 * mu * theta_new * (-1. / 3.)) -
+              (2 * mu * theta_new_b * (N_new(1) * N_new(0)));
+    D(1, 1) = volumetric_modulus + (2 * mu * theta_new * 2. / 3.) -
+              (2 * mu * theta_new_b * (N_new(1) * N_new(1)));
+    D(1, 2) = volumetric_modulus + (2 * mu * theta_new * (-1. / 3.)) -
+              (2 * mu * theta_new_b * (N_new(1) * N_new(2)));
+    D(1, 3) = -(2 * mu * theta_new_b * (N_new(1) * N_new(3)));
+    D(1, 4) = -(2 * mu * theta_new_b * (N_new(1) * N_new(4)));
+    D(1, 5) = -(2 * mu * theta_new_b * (N_new(1) * N_new(5)));
+
+    D(2, 0) = volumetric_modulus + (2 * mu * theta_new * (-1. / 3.)) -
+              (2 * mu * theta_new_b * (N_new(2) * N_new(0)));
+    D(2, 1) = volumetric_modulus + (2 * mu * theta_new * (-1. / 3.)) -
+              (2 * mu * theta_new_b * (N_new(2) * N_new(1)));
+    D(2, 2) = volumetric_modulus + (2 * mu * theta_new * 2. / 3.) -
+              (2 * mu * theta_new_b * (N_new(2) * N_new(2)));
+    D(2, 3) = -(2 * mu * theta_new_b * (N_new(2) * N_new(3)));
+    D(2, 4) = -(2 * mu * theta_new_b * (N_new(2) * N_new(4)));
+    D(2, 5) = -(2 * mu * theta_new_b * (N_new(2) * N_new(5)));
+
+    D(3, 0) = -(2 * mu * theta_new_b * (N_new(3) * N_new(0)));
+    D(3, 1) = -(2 * mu * theta_new_b * (N_new(3) * N_new(1)));
+    D(3, 2) = -(2 * mu * theta_new_b * (N_new(3) * N_new(2)));
+    D(3, 3) = mu * theta_new - (2 * mu * theta_new_b * (N_new(3) * N_new(3)));
+    D(3, 4) = -(2 * mu * theta_new_b * (N_new(3) * N_new(4)));
+    D(3, 5) = -(2 * mu * theta_new_b * (N_new(3) * N_new(5)));
+
+    D(4, 0) = -(2 * mu * theta_new_b * (N_new(4) * N_new(0)));
+    D(4, 1) = -(2 * mu * theta_new_b * (N_new(4) * N_new(1)));
+    D(4, 2) = -(2 * mu * theta_new_b * (N_new(4) * N_new(2)));
+    D(4, 3) = -(2 * mu * theta_new_b * (N_new(4) * N_new(3)));
+    D(4, 4) = mu * theta_new - (2 * mu * theta_new_b * (N_new(4) * N_new(4)));
+    D(4, 5) = -(2 * mu * theta_new_b * (N_new(4) * N_new(5)));
+
+    D(5, 0) = -(2 * mu * theta_new_b * (N_new(5) * N_new(0)));
+    D(5, 1) = -(2 * mu * theta_new_b * (N_new(5) * N_new(1)));
+    D(5, 2) = -(2 * mu * theta_new_b * (N_new(5) * N_new(2)));
+    D(5, 3) = -(2 * mu * theta_new_b * (N_new(5) * N_new(3)));
+    D(5, 4) = -(2 * mu * theta_new_b * (N_new(5) * N_new(4)));
+    D(5, 5) = mu * theta_new - (2 * mu * theta_new_b * (N_new(5) * N_new(5)));
+}
+
+void LinearJ2Plasticity3D::GetLawFeatures(Features& rFeatures)
+{
+    rFeatures.mOptions.Set(THREE_DIMENSIONAL_LAW);
+    rFeatures.mOptions.Set(INFINITESIMAL_STRAINS);
+    rFeatures.mOptions.Set(ISOTROPIC);
+    rFeatures.mStrainMeasures.push_back(StrainMeasure_Infinitesimal);
```

Comment:
push_back ? (just asking)

Reply:
It was copied from ElasticIsotropic3D (elastic_isotropic_3d.cpp, lines 180-200) and others, where push_back is used to store two values, StrainMeasure_Infinitesimal and StrainMeasure_Deformation_Gradient. But maybe you are right and we need to use a method.